File: MyFirstApp/app/src/main/java/mx/edu/cetys/garay/andrea/myfirstapp/MainActivity.java
package mx.edu.cetys.garay.andrea.myfirstapp;

import android.os.AsyncTask;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.View;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.StringRequest;
import com.android.volley.toolbox.Volley;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.BufferedInputStream;
import java.io.BufferedOutputStream;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.OutputStream;
import java.io.Reader;
import java.io.UnsupportedEncodingException;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;
import java.net.URL;
import java.util.Arrays;
import java.util.List;
import java.util.concurrent.ExecutionException;

public class MainActivity extends AppCompatActivity {
    private Gson gson;

    Button btnconsultar;
    EditText etId, etNombre, etTelefono;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        GsonBuilder gsonBuilder = new GsonBuilder();
        gsonBuilder.setDateFormat("M/d/yy hh:mm a");
        gson = gsonBuilder.create();
        btnconsultar = (Button) findViewById(R.id.btnConsultar);

        etId = (EditText) findViewById(R.id.etId);
        etNombre = (EditText) findViewById(R.id.etNombre);
        etTelefono = (EditText) findViewById(R.id.etTel);

        btnconsultar.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {
                try {
                    ConsultarDatos x = new ConsultarDatos();
                    String response = x.execute("http://138.68.231.116:5000/perfil").get();

                    JSONArray jsonArray = new JSONArray(response);
                    JSONObject jsonObject = jsonArray.getJSONObject(Integer.parseInt(etId.getText().toString()));
                    String nombre= jsonObject.getString("Nombre");
                    //String telefono= jsonObject.getString("Telefono");
                    String carrera = jsonObject.getString("Carrera");
                    //String direccion= jsonObject.getString("Direccion");

                    etNombre.setText(nombre);
                    etTelefono.setText(carrera);

                } catch (ExecutionException e) {
                    e.printStackTrace();
                } catch (InterruptedException e) {
                    e.printStackTrace();
                } catch (JSONException e) {
                    e.printStackTrace();
                }
            }
        });

        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                VolleyRequest();
                Snackbar.make(view, "Hola Amiguitos idgdeanos", Snackbar.LENGTH_LONG)
                        .setAction("Action", null).show();

            }
        });
    }


    private class ConsultarDatos extends AsyncTask<String, Void, String> {

        @Override
        protected String doInBackground(String... strings) {
            HttpURLConnection httpURLConnection = null;
            URL url = null;
            try {
                url = new URL(strings[0]);
            } catch (MalformedURLException e) {
                e.printStackTrace();
            }

            try {
                httpURLConnection = (HttpURLConnection)url.openConnection();
                httpURLConnection.connect();
                int code = httpURLConnection.getResponseCode();

                if(code == HttpURLConnection.HTTP_OK){
                    InputStream in = new BufferedInputStream(httpURLConnection.getInputStream());
                    BufferedReader reader = new BufferedReader(new InputStreamReader(in));
                    String line ="";
                    StringBuffer buffer = new StringBuffer();
                    while ((line=reader.readLine())!=null){
                        buffer.append(line);
                    }
                    return buffer.toString();
                }
            } catch (IOException e) {
                e.printStackTrace();
            }
            return null;
        }
        @Override
        protected void onPostExecute(String result){
            super.onPostExecute(result);
        }
    }



    public void VolleyRequest() {
        RequestQueue queue = Volley.newRequestQueue(this);
        String url = "http://138.68.231.116:5000/empresa";

        // Request a string response from the provided URL.
        StringRequest stringRequest = new StringRequest(Request.Method.GET, url,
                new Response.Listener<String>() {
                    @Override
                    public void onResponse(String response) {
                        List<Empresa> misEmpresas = Arrays.asList(gson.fromJson(response, Empresa[].class));
                        Toast.makeText(getApplicationContext(),
                                misEmpresas.toString(),
                                Toast.LENGTH_LONG).show();
                    }
                }, new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error) {
                Toast.makeText(getApplicationContext(), error.getMessage(), Toast.LENGTH_SHORT).show();
            }
        });

// Add the request to the RequestQueue.
        queue.add(stringRequest);
    }


    public void VolleyRequestFilter() {
        RequestQueue queue = Volley.newRequestQueue(this);
        String url = "http://138.68.231.116:5000/perfil/a4395238552679c9";

        // Request a string response from the provided URL.
        StringRequest stringRequest = new StringRequest(Request.Method.GET, url,
                new Response.Listener<String>() {
                    @Override
                    public void onResponse(String response) {
                        List<Usuario>  Usuarios = Arrays.asList(gson.fromJson(response, Usuario[].class));
                        Toast.makeText(getApplicationContext(),
                                Usuarios.toString(),
                                Toast.LENGTH_LONG).show();
                    }
                }, new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error) {
                Toast.makeText(getApplicationContext(), error.getMessage(), Toast.LENGTH_SHORT).show();
            }
        });

// Add the request to the RequestQueue.
        queue.add(stringRequest);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }





}

File: AppCetys/app/src/main/java/mx/edu/cetys/garay/andrea/appcetys/view/MainActivity.kt
package mx.edu.cetys.garay.andrea.appcetys.view

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.design.widget.Snackbar
import android.view.Menu
import android.view.MenuItem
import android.widget.Toast
import com.google.gson.Gson
import mx.edu.cetys.garay.andrea.appcetys.R
import mx.edu.cetys.garay.andrea.appcetys.controller.Utils
import com.android.volley.Request
import com.android.volley.Response
import com.android.volley.toolbox.StringRequest
import com.android.volley.toolbox.Volley
import kotlinx.android.synthetic.main.activity_main.*
import kotlinx.android.synthetic.main.content_main.*
import mx.edu.cetys.garay.andrea.appcetys.R.id.toolbar
import mx.edu.cetys.garay.andrea.appcetys.model.EndPoints
import mx.edu.cetys.garay.andrea.appcetys.model.Perfil

class MainActivity : AppCompatActivity() {
    private val utils = Utils()
    private val gson = Gson()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        setSupportActionBar(toolbar)

        fab.setOnClickListener { view ->
            Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                    .setAction("Action", null).show()
        }

        btn_login.setOnClickListener { _ ->
            obtenerPerfilRequest()
        }
    }

    private fun obtenerPerfilRequest() {
        val queue = Volley.newRequestQueue(this)

        val matricula = edt_matricula.text.toString()
        val password = edt_password.text.toString()

        val stringRequest = StringRequest(Request.Method.GET, EndPoints.PERFIL.url.toString(),
                Response.Listener { response ->
                    val perfiles = gson.fromJson(response, Array<Perfil>::class.java).toList()

                    val perfil = utils.buscarPerfilN(perfiles, matricula, password)
                    if (perfil != null) {
                        val intent = Intent(applicationContext, MainMenuActivity::class.java)
                        val intent2 = Intent(applicationContext, PerfilListViewActivity::class.java)
                        //intent.putExtra("nombre", perfil.nombre)
                        intent.putExtra("perfil", perfil)
                        startActivity(intent2)
                    } else {
                        Toast.makeText(
                                applicationContext,
                                "Perfil no encontrado, pague su inscripcion",
                                Toast.LENGTH_SHORT
                        ).show()
                    }

                }, Response.ErrorListener { error ->
            Toast.makeText(
                    applicationContext, error.message, Toast.LENGTH_SHORT).show()
        })
        queue.add(stringRequest)
    }


    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        return when (item.itemId) {
            R.id.action_settings -> true
            else -> super.onOptionsItemSelected(item)
        }
    }
}
